### ISSUE TITLE
proposal: apply penalty to features matched to autocomplete fragments

### ISSUE THREAD

**Issue by sbma44** [mapbox/carmen#521]
Haven't dug into how this might be implemented, but right now the greedy nature of autocomplete can lead to some lower-quality results. Hypothetically, consider:

`Smallville Co`

Assuming there are too many Smallvilles to consider them all and we don't get lucky with the ones we grab, this could easily wind up with two possible features with 0.49 relevance: some random Smallville and the country `Colombia`. We should prioritize features associated with completed terms so that ties like this one go to the definitely-finished parts of the query.

**Comment by apendleton** [comment]
FWIW I think this will be much easier to implement once degenless lands in carmen/carmen-cache. At the moment whether or not something is a full or partial match is not very well-preserved in the indexing process.

**Comment by sbma44** [comment]
ok, let's put this on hold pending degenlessness
